### PR TITLE
Cross Zamirski Model Training

### PR DESCRIPTION
**Pull request by MattsonCam**
This pr includes the cross zamirski model and the structure needed for training. It also include per-batch logging and removes irrelevant code. This code may change in the future to allow for training on Alpine due to the cuda memory constraint. As a result batch size has been reduced.

### CODE REVIEW

**Review by gwaybio — APPROVED** [review]

File: trainers/WGANGPTrainer.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+                    epoch=epoch,
+                    loss_mask=batch_data.get("loss_mask"),
+                )
+                generator_loss, generator_components = self._detach_components(
+                    generator_outputs
+                )
+
+                self.generator_optimizer.zero_grad()
+                generator_loss.backward()
+                self.generator_optimizer.step()
```
Comment (wli51) [inline trainers/WGANGPTrainer.py]: I don't remember the Cross-Zamirski trainer implementation that well, is equal number of update frequencies what they decided on. I believe in classical wGAN training the discriminator gets updated more frequently than the generator for stability.

**Review by wli51 — APPROVED** [review]
LGTM! Maybe the trainer should support differential stepping frequency between discriminator and generator?

File: trainers/WGANGPTrainer.py (hunk 1 above)
Comment (MattsonCam) [reply trainers/WGANGPTrainer.py]: I think you're right, I will update this. I know if degrades the loss contribution by normalize by the epoch

**Comment by MattsonCam** [comment]
Thanks for the reviews @gwaybio and @wli51 ! Merging now
